gpt_engineer/applications/cli/cli_step_bundle.py

```diff
@@ -0,0 +1,28 @@
+from gpt_engineer.core.ai import AI
+from gpt_engineer.core.code import Code
+from gpt_engineer.data.file_repository import FileRepository
+from gpt_engineer.core.step_bundle_interface import StepBundleInterface
+from gpt_engineer.core.default.steps import gen_code, gen_entrypoint, execute_entrypoint
+from gpt_engineer.core.default.paths import memory_path
+from gpt_engineer.applications.cli.learning import human_review
+
+
+class CliStepBundle(StepBundleInterface):
+
+    def __init__(self, workspace_path: str):
+        self.workspace_path = workspace_path
+        self.memory = FileRepository(memory_path(workspace_path))
```

Comment:
This looks a bit weird to me.

Why does stepbundleinterface know about the path?

Why is memory always on disk? While Code is in memory and version management is abstract..

Would be much more reasonable to pass in the memory object!

Reply:
Regarding this and the following comments:

I will make two abstract classes for memory and execution_environment and lean implementations of these will hold a path and keep the flow working roughly like it works now.